### ISSUE TITLE
Node.js env, cannot write the data grabbed from stdout as a proper gif image

### ISSUE THREAD

**Issue by lnshi** [kohler/gifsicle#119]
const {execFile} = require('child_process');

    const child = execFile(
      gifsicle,
      ['-', '--colors=256', `--resize-fit=${width}x${height}`, '--resize-method=sample']
    );

    child.stdin.write(require('fs').readFileSync(myOriginalGifPath));

    child.stdout.on('data', (data) => {
      require('fs').writeFile('leo.gif', data, () => {

      });;
    });

    child.stdout.on('end', () => {

    });

    child.stdin.end();

As above code, if I just use the option `-o / --output dest path`, then it works perfectly fine, but if I write the data grabbed from the stdout by myself with the Node.js `writeFile` API, then it is not a proper gif anymore, tried various ways, cannot make it work, pls help.

**Comment by h3xx** [comment]
Can you provide a specific input GIF file it doesn't work for, or is it all GIFs?

Also, what do you mean by "not a proper gif anymore" - does it produce a broken file? If so, can you provide that file?

**Comment by lnshi** [comment]
@h3xx  I tested with this [gif](https://media.giphy.com/media/IB9foBA4PVkKA/giphy.gif), and I guess it will be same for all, coz the wrongly produced gif file size is a lot bigger than the correct one: 11kb vs 4kb like that.

`not proper gif anymore` means my Mac `preview` cannot load it properly, keep loading but cannot show/play it correctly, and Google chrome also cannot.

Eh, if I change the file extension to `.txt`, then compare the correct one and the wrong one in text editor, the result kindly like [this](https://stackoverflow.com/questions/12868089/nodejs-write-binary-buffer-into-a-file): the correct one will be some light clear characters, but the wrong one is all bold characters.

**Comment by kohler** [comment]
This is almost certainly not an error with gifsicle. It is an error with your node program. Most likely you are not using the `{encoding: "binary"}` on your stdin stream.

**Comment by lnshi** [comment]
@kohler I wasn't saying it is gifsicle problem, I am looking for help. I tried various of encoding: binary, even utf8, iso8559-1 etc.

If I set the encoding before I read the file and send to gifsicle then even the -o / --outpu can't work. If I set encoding for the writeFile, all cannot produce a proper gif image.

More clear to say: I want to find out what gifsicle send into the stdout, and how can I write the data to file properly.
